src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/AggregateContainsRelationCollection.cs

```diff
@@ -0,0 +1,190 @@
+                // into this object to read partially constructed state which is valid and must be supported.
+                _isMaterialized = true;
+
+                OnStateUpdated();
+            }
+        }
+
+        internal void RaiseChange(NotifyCollectionChangedEventArgs e) => CollectionChanged?.Invoke(this, e);
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            foreach (AggregateContainsRelationCollectionSpan span in _spans)
+            {
+                if (span.Items != null)
+                {
+                    foreach (IRelatableItem item in span.Items)
+                    {
+                        yield return item;
+                    }
+                }
+            }
+        }
+
+        object IList.this[int index]
+        {
+            get
+            {
+                if (_isMaterialized)
+                {
+                    foreach (AggregateContainsRelationCollectionSpan span in _spans)
+                    {
+                        if (span.Items == null)
+                        {
+                            continue;
+                        }
+
+                        int spanCount = span.Items.Count;
+
+                        if (spanCount == 0)
+                        {
+                            continue;
+                        }
+
+                        if (index < spanCount)
+                        {
+                            return span.Items[index];
+                        }
+
+                        index -= spanCount;
+                    }
+                }
+
+                throw new ArgumentOutOfRangeException(nameof(index), index, "Invalid index.");
+            }
+            set => throw new InvalidOperationException();
+        }
+
+        int IList.IndexOf(object value)
+        {
+            if (_isMaterialized && value is IRelatableItem item)
+            {
+                int baseIndex = 0;
+
+                foreach (AggregateContainsRelationCollectionSpan span in _spans)
+                {
+                    if (span.Items == null)
+                    {
+                        continue;
+                    }
+
+                    int spanCount = span.Items.Count;
+
+                    if (spanCount == 0)
+                    {
+                        continue;
+                    }
+
+                    int index = span.Items.IndexOf(item);
+
+                    if (index != -1)
+                    {
+                        return baseIndex + index;
+                    }
+
+                    baseIndex += spanCount;
+                }
+            }
+
+            return -1;
+        }
+
+        bool IList.Contains(object value) => value is IRelatableItem item && _spans.Any(span => span.Items?.Contains(item) == true);
+
+        void ICollection.CopyTo(Array array, int index) => throw new InvalidOperationException();
```

Comment:
Why?

Reply:
Because it's not needed by `ListCollectionView` and implementation is non-trivial.